src/Components/Components/src/ElementReference.cs

```diff
@@ -23,13 +23,24 @@ namespace Microsoft.AspNetCore.Components
         /// </remarks>
         public string Id { get; }
 
-        public ElementReference(string id)
+        /// <summary>
+        /// Gets the <see cref="IServiceProvider"/> instance.
+        /// </summary>
+        public IServiceProvider ServiceProvider { get; }
+
+        /// <summary>
+        /// Instantiates a new <see cref="ElementReference" />.
+        /// </summary>
+        /// <param name="id">A unique identifier for this <see cref="ElementReference"/>.</param>
+        /// <param name="serviceProvider">The <see cref="ElementReference"/>'s <see cref="IServiceProvider"/>.</param>
+        public ElementReference(string id, IServiceProvider serviceProvider)
```

Comment:
@dotnet/aspnet-blazor-eng isn't this a breaking change? Is there a scenario where customers may have been using the old constructor in their code?
While necessary, I'm wondering whether we should also post an announcement about this.

Reply:
I guess we could keep the old ctor to avoid the breaking change. It's not particularly common to new up these instances though.

Reply:
It's a major version, and I customers don't create this type themselves, so it's fine to break it. The major issues are unit tests that people might have written.

Reply:
Retaining the old constructor with `[Obsolete]` would be OK as it continue to work fine in unit tests that don't need to read back the service provider (or whatever ends up getting stored there).

However if an `[Obsolete]` also requires us to post an announcement, maybe it's easier just to do a regular breaking change.